### ISSUE TITLE
[Analyzer Suggestion] Use target-typed new to simplify fields initialization

### ISSUE THREAD

**Issue by Youssef1313** [dotnet/roslyn#47189]
**Brief description:**

Use "target-typed new", the new C# 9 feature, to simplify fields initialization

**Languages applicable:**

C#

**Code example that the analyzer should report:**

```csharp
public class SomeClass
{
    private readonly AnotherClass _anotherClass = new AnotherClass(......);
}
```

The "AnotherClass" part after "new" should be faded as unnecessary and the codefix should remove it.

**Additional information:**

Applicable for C# 9 and later only.

**Comment by sharwell** [comment]
Duplicate of #46700
